config/common.go

```diff
@@ -141,3 +174,95 @@ func (cs *commonStore) validate(cfg *model.Config) error {
 
 	return nil
 }
+
+// removeEnvOverrides returns a new config without the current environment overrides.
+// If a config variable has an environment override, that variable is set to the value that was
+// read from the store.
+func (cs *commonStore) removeEnvOverrides(cfg *model.Config) *model.Config {
```

Comment:
Given your pending work on a `mergeConfig(cfg1, cfg2 *model.Config) *model.Config`, I'm wondering if we could reuse that method here instead. It would mean having to represent `*model.Config` of /only/  the values that were /not/ overridden, so that we could force them back on before `persist`. 

Maybe building /that/ struct amounts to the same amount of work, so 0/5 right now, given that this implementation detail is entirely abstracted within the package.

Reply:
I investigated using the merge, but in the end I think it might be too complicated. First, you're right -- it basically is the same amount of work to create a blank config with only the envOverrides set. But then we'd also have to be very sure that none of the values in an empty config struct are set to their zero values (and I noticed that not /every/ variable is a pointer, which means we'd want to check if they had changed and keep track before/after the merge, just to be safe).